EIPS/eip-1231.md

```diff
@@ -0,0 +1,39 @@
+---
+eip: 1231
+title: Reduce cost of CALLs to precompiles
+author: Matthew Di Ferrante (@mattdf)
+status: Draft
+type: Standards Track
+category: Core
+created: 2018-07-19
+requires: 150, 158
+---
+
+## Short Description
+
+The Spurious Dragon hard fork increased the cost of CALLs to mitigate protocol DoS attacks that involved reads, but it failed to distinguish between CALLs to precompiles and CALLs to deployed contracts. This is an oversight that should be corrected.
+
+## Motivation
+
+There are useful precompiles such as a ECADD, SHA256, RIPEMD and the identity precompile (useful for memcpy) which have a gas cost that is far below or approximately equal to the cost of a CALL, which restricts their usefulness if they need to be called many times in a loop. This can cause a 2x to 10x increase in gas cost due to the CALL cost overhead, particularly with SHA256 and ECADD which often require tight loops for certain cryptographic protocols.
+
+In the Spurious Dragon hard fork, the cost of CALLs across the board was raised to 700 gas to mitigate DoS from state reads, without making a distinction between precompiles and deployed contracts. It does not make sense that precompiles should also suffer from the DoS mitigation penalty, since the precompiles are part of the core node code and hence never need to be fetched from disk. The CALL cost for precompiles should be reverted back to pre-EIP150 values, to 40 gas per call.
+
+## Specification
+
+Following is a table with the current gas cost and new gas cost:
+
+| Opcode name                    | Opcodes   | Current Gas Cost               | Updated Gas Cost    |
+| ------------------------------ | --------- | -----------------------------  | ------------------- |
+| `CALL` (to precompiles)        | `0xf1`    | 700<sup>[1]</sup>              | 40                  |
```

Comment:
Is the table necessary, since it's the same for all types of call?

Reply:
I think it can be removed..